app/uk/gov/hmrc/helloworldupscan/repository/UserSessionRepository.scala

```diff
@@ -91,7 +90,7 @@ class UserSessionRepository @Inject() (mongoComponent: ReactiveMongoComponent)(i
           status -> Json.toJson(newStatus)
         )
       ), upsert = true)) yield {
-      result.result[UploadStatus].getOrElse(throw new Exception("Update failed, no document modified"))
+      result.result[UploadDetails].map(_.status).getOrElse(throw new Exception("Update failed, no document modified"))
```

Comment:
This was the crucial change to make things work.

Reply:
There may not be any reason to even return the value that was set back? `Future[Unit]` could be sufficient.